### DISCUSSION TITLE
Formula `gdal` is missing dependency `sfcgal`

### DISCUSSION THREAD

**Discussion by ping13** [Homebrew/discussions#5709] [Writing Formulae/Casks]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.4.3-85-g4d0deeb
ORIGIN: https://github.com/Homebrew/brew
HEAD: 4d0deeb68e488fd2208c8a1058c506e70b182a63
Last commit: 14 hours ago
Core tap JSON: 03 Nov 20:58 UTC
Core cask tap JSON: 03 Nov 20:58 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.YAcZcTA7sC/org.xquartz:0
HOMEBREW_MAKE_JOBS: 6
HOMEBREW_SORBET_RUNTIME: set
Homebrew Ruby: 3.3.5 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/3.3.5/bin/ruby
CPU: hexa-core 64-bit kabylake
Clang: 15.0.0 build 1500
Git: 2.47.0 => /usr/local/bin/git
Curl: 8.7.1 => /usr/bin/curl
macOS: 14.7-x86_64
CLT: 15.3.0.0.1.1708646388
Xcode: N/A
```

### Output of `brew doctor`

```shell
Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  openssl@1.1

Warning: You have unlinked kegs in your Cellar.
Leaving kegs unlinked can lead to build-trouble and cause formulae that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
  python-packaging
  certifi
```

### Description of issue

The formula [gdal](https://github.com/Homebrew/homebrew-core/blob/d201a8d6fb68878467aed794ae5fa6e5eb54474f/Formula/g/gdal.rb) is missing a dependency [sfcgal](https://formulae.brew.sh/formula/sfcgal).

For example, this means that for some format transformation with `ogr2ogr`, you get errors like:

```
ERROR 6: SFCGAL support not enabled.
```

If you include   

```
depends_on "sfcgal"
```
in the formula `gdal`, the error disappeared and SFCGAL is enabled

I validated this by using an edited local formula and build goal from source (see this [gist](https://gist.github.com/ping13/c736388e214b51f7e25174502e1420f7)). I would do a PR, but it wasn't really clear to me what I'd need todo and it's late in the evening ;-).

**Comment by gromgit** [comment]
If the [official PR instructions](https://docs.brew.sh/How-To-Open-a-Homebrew-Pull-Request) aren't clear, please ask questions and/or suggest improvements. Alas, English is far from being a universal instruction set. 🤷‍♂️
